### PR TITLE
Written unit testing for methods of components of test and reports module.

### PR DESCRIPTION
**Pull request by Madhurima94**
**Fixed Issues**

#24194,#24268

**Files Changed**
individual-report.component.spec.ts-written unit testing for methods of individual report component.
tests-dashboard.component.spec.ts-Written unit testing for methods of test dashboard component.
test-view.component.spec.ts-Written unit testing for methods of test view component.
delete-test-dialog.component.spec.ts-Written unit testing for methods of delete test dialog component.
duplicate-test-dialog.component.spec.ts-Written unit testing for methods of duplicate test dialog component.
test-settings.component.spec.ts-Written unit testing for methods of test settings component.
karma.conf.js-Defined clipboard,popover,ng2-charts and ngx-clipboard paths.
create-test-header.component.spec.ts-Written unit testing for methods of create-test-header component.
create-test-footer.component.spec.ts-Written unit testing for methods of create-test-footer component.

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/554)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export class TestSettingsComponent implements OnInit {
     isIpAddressAdded: boolean;
     isIpAddressFieldNull: boolean;
     
-    constructor(public dialog: MdDialog, private testService: TestService, private router: Router, private route: ActivatedRoute, private snackbarRef: MdSnackBar) {
+    constructor(public dialog: MdDialog, private testService: TestService, private router: Router, private route: ActivatedRoute, public snackbarRef: MdSnackBar) {
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts]: why public?

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-view/test-view.component.ts [hunk 2]
```diff
@@ -40,8 +40,10 @@ export class TestViewComponent implements OnInit {
     tooltipMessage: string;
     isCategorySelected: boolean;
     disablePreview: boolean;
+    duplicateTestDialogData: DuplicateTestDialogComponent;
+    deleteTestDialogData: DeleteTestDialogComponent;
 
-    constructor(public dialog: MdDialog, private testService: TestService, public snackBar: MdSnackBar, private router: Router, private route: ActivatedRoute) {
+    constructor(public dialog: MdDialog, private testService: TestService, public snackBar: MdSnackBar, private router: Router, public route: ActivatedRoute) {
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-view/test-view.component.ts]: same

Comment (Madhurima94) [reply Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-view/test-view.component.ts]: It need not be public. Will have to change it.

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts (hunk 1 above)
Comment (Madhurima94) [reply Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts]: It has to be public in order to test whether the snackbar in the test-settings page has been opened or not. On being private, it cannot be accessed from the spec file.

**Pull request merged**